File: Client/src/app/models/project.ts
import { StatusEnum } from "../enums/status-enum";
import { Group } from "./group";

export interface Project {
    id: number;
    projectNumber: number;
    name: string;
    customer: string;
    status: StatusEnum;
    startDate: Date;
    endDate: Date;
    visas: number[];
}

File: Server/src/PIMTool.Core/Domain/Entities/Project.cs
using PIMTool.Core.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace PIMTool.Core.Domain.Entities
{
    public class Project : BaseEntity
    {
        public decimal ProjectNumber { get; set; }
        public string Name { get; set; } = null!;
        public string Customer {  get; set; } = null!;
        public ProjectStatus.EnumStatus Status { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal GroupId { get; set; }
        public Group? Group { get; set; }
        public ICollection<ProjectEmployee>? ProjectEmployees { get; set; }
    }
}